sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java

```diff
@@ -98,8 +102,23 @@ public boolean downloadToFile(MavenLocation location, File toFile) {
         LOG.info("Found {} at {}", location, url);
         return Optional.of(toFile);
       } catch (HttpException e) {
-        if (e.getCode() != 404 && e.getCode() != 401 && e.getCode() != 403) {
+        if (e.getCode() != HTTP_NOT_FOUND && e.getCode() != HTTP_UNAUTHORIZED && e.getCode() != HTTP_FORBIDDEN) {
           throw new IllegalStateException("Failed to request " + url, e);
+        } else {
+          String errorMessage;
+          try {
+            JsonArray errors = Json.parse(e.getBody()).asObject().get("errors").asArray();
```

Comment:
The error may to be json

Reply:
Can you clarify, what do you mean?

Reply:
(sorry i missclicked the "resolve" button). Yeah so i mean that the payload return by the server may be non-json (like plain text). In that case, you will have a Json Parse Exception

Reply:
aaaaand i realise that there is a catch for that :D so, nevermind, you did just fine :D